src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -62,6 +68,46 @@ public void ThrowOnErrors()
             }
         }
 
+        /// <summary>
+        /// Gets the type of a syntax node.
+        /// </summary>
+        /// <param name="node"></param>
+        /// <returns></returns>
+        public FormulaType GetNodeType(TexlNode node) => FormulaType.Build(_binding.GetType(node));
+
+        /// <summary>
+        /// Whether the node is a constant node.
+        /// </summary>
+        /// <param name="node"></param>
+        /// <returns></returns>
+        public bool IsNodeConstant(TexlNode node) => _binding.IsConstant(node);
+
+        /// <summary>
+        /// Check whether function invocation in type-correct.
+        /// </summary>
+        /// <param name="fnc"></param>
+        /// <param name="nodes"></param>
+        /// <param name="types"></param>
+        /// <param name="retType"></param>
+        /// <returns></returns>
+        public bool CheckInvocation(FunctionInfo fnc, IReadOnlyList<TexlNode> nodes, IReadOnlyList<FormulaType> types, out FormulaType retType)
```

Comment:
> IReadOnlyList<FormulaType> types [](http://example.com/codeflow?start=84&length=33)

Could we remove these and get from the nodes?

Rename nodes to args?

Reply:
Removed types and renamed to args.

Reply:
Removed for now altogether.